src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -107,12 +107,41 @@ void OTARequestor::InitState(intptr_t context)
     OTARequestor * requestorCore = reinterpret_cast<OTARequestor *>(context);
     VerifyOrDie(requestorCore != nullptr);
 
-    // This results in the initial periodic timer kicking off
-    requestorCore->RecordNewUpdateState(OTAUpdateStateEnum::kIdle, OTAChangeReasonEnum::kSuccess);
+    if (requestorCore->mCurrentUpdateState != OTAUpdateStateEnum::kApplying)
```

Comment:
I think we should add here `|| !IsFirstImageRun`. Otherwise, if a new image crashes and is reverted, we will switch to `kApplying` state, but the driver will never confirm the image.

Reply:
`IsFirstImageRun` is in image processor which OTARequestor does not have access to. Instead, I addressed this by modifying `GenericOTARequestorDriver::Init` to prevent problems with the new image from causing  the "stuck in the `kApplying` state" scenario. Specifically, I reset the OTA states either when confirm fails or if this is `!IsFirstImageRun` but we're also not in the idle state.

Reply:
`OTARequestor` does have access to `OTAImageProcessor`, but I think your solution is cleaner now.